### PR TITLE
Passive scan script for finding potential s3 Bucket URLs

### PR DESCRIPTION
**Pull request by alishasinghania**
Closed previously raised pull request by mistake. Create a new one with single commit as suggested in previous comment.

### CODE REVIEW

File: passive/s3.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// S3 bucket finder by alishasinghania09@gmail.com 
+
+function scan(ps, msg, src) {
+    // populate some parameters which will be needed if s3 bucket url is present
+    var alertRisk = 1
+    var alertReliability = 3
```
Comment (kingthorin) [inline passive/s3.js:6]:
```suggestion
    var alertConfidence= 3
```

I'm sure you had copied this from other examples, we're trying to get things cleaned up.

If you are/were having problems with squashing or amending commits just let me know I don't mind doing the change for you.

File: passive/s3.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        var body = msg.getResponseBody().toString()
+        if (re.test(body)) {
+            re.lastIndex = 0 
+            var founds3bucket = []
+            var buckets
+            while (buckets = re.exec(body)) {
+                founds3bucket.push(buckets[0]);
+            }
+            //raise the alert
+            ps.raiseAlert(alertRisk, alertReliability, alertTitle, alertDesc, url, '', '', founds3bucket.toString(), alertSolution, '', cweId, wascId, msg);
```
Comment (kingthorin) [inline passive/s3.js:36]:
```suggestion
            ps.raiseAlert(alertRisk, alertConfidence, alertTitle, alertDesc, url, '', '', founds3bucket.toString(), alertSolution, '', cweId, wascId, msg);
```

**Review by kingthorin — CHANGES_REQUESTED** [review]

File: passive/s3.js (hunk 1 above)
Comment (thc202) [reply passive/s3.js:6]: It was suggested in the previous review too https://github.com/zaproxy/community-scripts/pull/213#discussion_r602141056

**Comment by alishasinghania** [comment]
@kingthorin I made the changes and renamed the reliability to confidence. I am having problem in squashing the commit. Can you help in that?

**Comment by kingthorin** [comment]
Now I messed it up 😭 

Sec, I'll add instructions.

**Comment by alishasinghania** [comment]
should I raise it again ?

**Comment by kingthorin** [comment]
Just a sec, yes but let me get you some instructions.

**Comment by kingthorin** [comment]
The following assume you're in your local community-scripts repo/directory:
```console
git checkout -b s3
<add the file ... content below if you've lost track>
git commit -a -m "Passive scan script for finding potential s3 Bucket URLs" -m "Closed previously raised pull requests by mistake. Create a new one with single commit as suggested in previous comments." --signoff
git push -u origin s3
<goto GitHub web interface and create your PR>
```

<details>
<summary>Current s3.js code <sub>(Click the triangle/control to the left to expand</sub></summary>

```js
// S3 bucket finder by alishasinghania09@gmail.com 

function scan(ps, msg, src) {
    // populate some parameters which will be needed if s3 bucket url is present
    var alertRisk = 1
    var alertConfidence= 3
    var alertTitle = 's3 Bucket URL'
    var alertDesc = 's3 Bucket URL found in response.'
    var alertSolution = 'Remove s3 Buckets name from response or make sure the permissions in bucket are configured properly.'
    var cweId = 200
    var wascId = 13

    // the regex for s3 bucket url and it must appear within /( and )/g
    var re = /((s3:\\[a-zA-Z0-9-\.\\_]+)|((s3-|s3\.)?(.*)\.amazonaws\.com))/g

    // we need to set the url variable to the request or we cant track the alert later
    var url = msg.getRequestHeader().getURI().toString();

    // If the file type is image jpeg/png , then the scan will be skipped
    var contenttype = msg.getResponseHeader().getHeader("Content-Type")
    var unwantedfiletypes = ['image/png', 'image/jpeg','image/gif','application/x-shockwave-flash','application/pdf']
    if (unwantedfiletypes.indexOf(""+contenttype) >= 0) {
        return
    }
    else{
        // test the regex against the message body
        var body = msg.getResponseBody().toString()
        if (re.test(body)) {
            re.lastIndex = 0 
            var founds3bucket = []
            var buckets
            while (buckets = re.exec(body)) {
                founds3bucket.push(buckets[0]);
            }
            //raise the alert
            ps.raiseAlert(alertRisk, alertConfidence, alertTitle, alertDesc, url, '', '', founds3bucket.toString(), alertSolution, '', cweId, wascId, msg);
        }
    }
}
```

Thanks, sorry for messing it up (I seem to do that every time someone creates a PR from their master branch 😞 ).

**Pull request closed without merging**